cmd/prometheus-config-reloader/main.go

```diff
@@ -72,24 +99,55 @@ func main() {
 	}
 
 	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout))
+
 	if *logFormat == logFormatJson {
 		logger = log.NewJSONLogger(log.NewSyncWriter(os.Stdout))
 	}
+
+	switch *logLevel {
+	case logLevelDebug:
+		logger = level.NewFilter(logger, level.AllowDebug())
+	case logLevelWarn:
+		logger = level.NewFilter(logger, level.AllowWarn())
+	case logLevelError:
+		logger = level.NewFilter(logger, level.AllowError())
+	default:
+		logger = level.NewFilter(logger, level.AllowInfo())
+	}
+
 	logger = log.With(logger, "ts", log.DefaultTimestampUTC)
 	logger = log.With(logger, "caller", log.DefaultCaller)
 
 	if createStatefulsetOrdinalFrom != nil {
 		if err := createOrdinalEnvvar(*createStatefulsetOrdinalFrom); err != nil {
-			logger.Log("msg", fmt.Sprintf("Failed setting %s", statefulsetOrdinalEnvvar))
+			level.Warn(logger).Log("msg", fmt.Sprintf("Failed setting %s", statefulsetOrdinalEnvvar))
 		}
 	}
 
-	logger.Log("msg", fmt.Sprintf("Starting prometheus-config-reloader version '%v'.", version.Version))
+	level.Info(logger).Log("msg", fmt.Sprintf("Starting prometheus-config-reloader version '%v'.", version.Version))
+
+	r := prometheus.NewRegistry()
+	r.MustRegister(
+		prometheus.NewGoCollector(),
+		prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}),
+	)
 
 	var g run.Group
 	{
 		ctx, cancel := context.WithCancel(context.Background())
-		rel := reloader.New(logger, *reloadURL, *cfgFile, *cfgSubstFile, *rulesDir)
+		rel := reloader.New(
+			logger,
+			r,
+			&reloader.Options{
+				ReloadURL:     *reloadURL,
+				CfgFile:       *cfgFile,
+				CfgOutputFile: *cfgSubstFile,
+				WatchedDirs:   *watchedDir,
+				DelayInterval: 1 * time.Second,
+				WatchInterval: 3 * time.Minute,
```

Comment:
Can you explain a bit the numbers here you used as the defaults. And is there a way to override them I couldn't see it yet? Thanks!

Reply:
The `WatchInterval` value is what was hardcoded previously in the Thanos code. I've chosen the `DelayInterval` value arbitrarily but open to make both values configurable.

Reply:
Happy to have it in a followup or if we run into problems or per user requests, was just curious of the values. Up to you!

Reply:
suggestion: maybe let's also add comments rationalizing the chosen numbers

Reply:
I've added a comment for each `const` identifier.